lib/kinesis/consumer.rb

```diff
@@ -81,32 +81,27 @@ def setup_shards
 
       response.shards.each do |shard_data|
         shard_id = shard_data[:shard_id]
+        shard_locked = @state.lock_shard(shard_id, Time.now + LOCK_DURATION)
 
-        lock_shard(shard_id)
-
-        create_shard_reader(shard_id) unless @shards.key?(shard_id)
+        if shard_locked
```

Comment:
Noob question - what situation would there be unlocked shard? If it was locked by another thread but got released in-between L84 and L86?

Reply:
Not so much unlocked as failed to lock:
1. when the current consumer is not the one who has the latest lock
2. when an error is encountered trying to lock the shard (e.g. dynamodb issue)